crates/apub/src/protocol/objects/person.rs

```diff
@@ -39,6 +39,7 @@ pub struct Person {
   pub(crate) public_key: PublicKey,
   pub(crate) published: Option<DateTime<FixedOffset>>,
   pub(crate) updated: Option<DateTime<FixedOffset>>,
+  pub(crate) ban_expires: Option<DateTime<FixedOffset>>,
```

Comment:
I never thought about federating site bans as part of the `Person`. But for it to make sense, you probably also have to add a field `banned` to support permanent bans. I'm not sure how useful this is though, because anyway we will have to federate site bans with a `Block` activity in order to:

- Remove all existing posts/comments by the user (according to checkbox)
- Write to mod log
- Make it federate immediately, instead of waiting for actor refetch (up to 24h)

So its not a bad idea to do this, but once the activity is implemented, its usefulness will be very limited.

Reply:
In that case I'll just remove this field for now, and we can include think about how to handle temp bans in #1650

Reply:
You can add it to the `BlockUserFromCommunity` activity as i described in my comment above, then temporary community bans will work. Only site bans federation needs to be implemented still (both temporary and permanent).

Reply:
mmk I *think* I got it, let me know if that's the right way.